detectable/src/main/java/com/synopsys/integration/detectable/detectables/poetry/parser/ToolPoetrySectionParser.java

```diff
@@ -85,6 +85,17 @@ private void processKeyForRootPackages(TomlParseResult parseResult, PoetryOption
     }
 
     private void addAllTableKeysToSet(Set<String> set, TomlTable table) {
-        set.addAll(table.dottedKeySet());
+        for (String key : table.dottedKeySet()) {
+            if (key.equalsIgnoreCase(PYTHON_COMPONENT_NAME))
+                continue;
+
+            int dotIndex = key.indexOf('.', 0);
+            if (dotIndex == -1) {
+                set.add(key);
+            } else {
+                // remove the ".extras" part
```

Comment:
When testing a project I discovered that when a dependency is specified as `uvicorn = {extras = ["standard"], version = "^0.20.0"}`, the key here contains `.extras`. Since the Poetry detector does not deal with extras in any manner right now, I decided to simply continue to ignore extras.

We may want to revisit this if we ever want to implement inspection of extras in Poetry.

Reply:
What exactly does this ".extras" part mean?

Reply:
Some general information on "extras" in Poetry: https://python-poetry.org/docs/pyproject/#extras

The Poetry Detectable always seems to have ignored extras, and I am planning to make a ticket to resolve that.
Since we're now parsing the `pyproject.toml` in more detail we're encountering extras in a new place. For now, I am choosing to ignore them as it is done when parsing the lock file.

When parsing `pyproject.toml` that contains something like this:

```
[tool.poetry.dependencies]
python = "^3.11"
fastapi = "^0.109.2"
uvicorn = {extras = ["standard"], version = "^0.20.0"}
```

One of the toml keys that the parser provides is something like `unicorn.extras` and `uvicorn` on its own doesn't seem to be returned. So, to ensure we ignore the "extras" bit but still add the `unicorn` dependency, I am stripping out the `extras` bit.